src/render/opengl/SDL_render_gl.c

```diff
@@ -1916,19 +1916,12 @@ GL_CreateRenderer(SDL_Window * window, Uint32 flags)
                 data->shaders ? "ENABLED" : "DISABLED");
 #if SDL_HAVE_YUV
     /* We support YV12 textures using 3 textures and a shader */
-    if (data->shaders && data->num_texture_units >= 3) {
-        renderer->info.texture_formats[renderer->info.num_texture_formats++] = SDL_PIXELFORMAT_YV12;
-        renderer->info.texture_formats[renderer->info.num_texture_formats++] = SDL_PIXELFORMAT_IYUV;
-    }
+    if (!data->shaders || data->num_texture_units < 3)
+        renderer->info.num_texture_formats -= 2;
```

Comment:
This feels a bit fragile for a couple of reasons:
- It wouldn't allow for potential future cases where there are two separate extensions that provide different formats.
- At some point, I'd like to get the software renderer to optimise the order of the format list based on the format used by the underlying framebuffer, which would require the ability to modify the list of formats at runtime.

Reply:
True. I just wish that I could think of a solution that doesn't involve memory allocation or giving each copy of the struct its own arbitrarily-sized buffer (which was the cause of this limitation in the first place).